unicodetools/src/main/java/org/unicode/xml/AttributeResolver.java

```diff
@@ -0,0 +1,458 @@
+                                && resolvedValue.startsWith("CJK UNIFIED IDEOGRAPH-")) {
+                            return "CJK UNIFIED IDEOGRAPH-#";
+                        }
+                        if (resolvedValue != null
+                                && resolvedValue.startsWith("CJK COMPATIBILITY IDEOGRAPH-")) {
+                            return "CJK COMPATIBILITY IDEOGRAPH-#";
+                        }
+                        if (resolvedValue != null
+                                && resolvedValue.startsWith("TANGUT IDEOGRAPH-")) {
+                            return "TANGUT IDEOGRAPH-#";
+                        }
+                        if (resolvedValue != null
+                                && resolvedValue.startsWith("KHITAN SMALL SCRIPT CHARACTER-")) {
+                            return "KHITAN SMALL SCRIPT CHARACTER-#";
+                        }
+                        if (resolvedValue != null && resolvedValue.startsWith("NUSHU CHARACTER-")) {
+                            return "NUSHU CHARACTER-#";
+                        }
+                        if (resolvedValue != null
+                                && resolvedValue.startsWith("EGYPTIAN HIEROGLYPH-")) {
+                            return "EGYPTIAN HIEROGLYPH-#";
+                        }
+                        return Optional.ofNullable(resolvedValue).orElse("");
+                    case kDefinition:
+                        return resolvedValue;
+                    default:
+                        if (resolvedValue != null) {
+                            return resolvedValue.replaceAll("\\|", " ");
+                        }
+                        return "";
+                }
+            case Catalog:
+                switch (prop) {
+                    case Age:
+                        String age = map_age.get(codepoint).getShortName();
+                        return (age.equals("NA")) ? "unassigned" : age;
+                    case Block:
+                        return map_block.get(codepoint).getShortName();
+                    case Script:
+                        return map_script.get(codepoint).getShortName();
+                    case Script_Extensions:
+                        StringBuilder extensionBuilder = new StringBuilder();
+                        String[] extensions = map_script_extensions.get(codepoint).split("\\|", 0);
+                        for (String extension : extensions) {
+                            extensionBuilder.append(
+                                    UcdPropertyValues.Script_Values.valueOf(extension)
+                                            .getShortName());
+                            extensionBuilder.append(" ");
+                        }
+                        return extensionBuilder.toString().trim();
+                    default:
+                        throw new RuntimeException("Missing Catalog case");
+                }
+            case Enumerated:
```

Comment:
This (and the associated pile of `UnicodeMap`s) seems like it is going to be a bit annoying to maintain as we add properties.
Is there a reason why you are not doing something like
```java
final UnicodeProperty property = indexUnicodeProperties.getProperty(prop);
final List<String> valueAliases = property.getValueAliases(property.getValue(codepoint));
return valueAliases.size() == 1 ? valueAliases.get(0) : valueAliases.get(1);
```
for most of them (special-casing Decomposition_Type etc. as needed)?

Reply:
I think that I was assuming that there were going to be more special cases, but I agree that as there are not, your solution is better. Implemented.